src/main/java/io/cryostat/net/web/http/api/v2/graph/TargetNodesFetcher.java

```diff
@@ -110,6 +110,13 @@ public List<TargetNode> getAuthenticated(DataFetchingEnvironment environment) th
                             .filter(n -> Objects.equals(n.getName(), nodeName))
                             .collect(Collectors.toList());
         }
+        if (filter.contains(FilterInput.Key.NAMES)) {
+            List<String> names = filter.get(FilterInput.Key.NAMES);
+            result =
+                    result.stream()
+                            .filter(n -> names.contains(n.getName()))
```

Comment:
There is another possibility here that I was considering.

Rather than having the `names` filter be a list of exact string matches that the given node's name must be a member of, this could reuse the [LabelSelectorMatcher](https://github.com/cryostatio/cryostat/blob/main/src/main/java/io/cryostat/net/web/http/api/v2/graph/labels/LabelSelectorMatcher.java). I will give a bit more thought to which approach I would prefer we go for here. Regardless, I suggest you take a look at that class and its unit tests and familiarize yourself with it.

Reply:
Maybe a nice flexible way to do this would be:

1. Accept the `name` filter for a single exact match. This is the same thing as `names: ['one-match']` and already done.
2. Accept the `names` filter for a list of exact matches. This is what you have done here.
3. Accept a `nameMatch` filter that accepts the `LabelSelectorMatcher` syntax.

Reply:
for now do i need to do any changes?

Reply:
https://github.com/cryostatio/cryostat/issues/1552